common/src/jni/main/cpp/conscrypt/native_crypto.cc

```diff
@@ -6877,6 +6877,25 @@ static jlong NativeCrypto_SSL_clear_options(JNIEnv* env, jclass, jlong ssl_addre
     return result;
 }
 
+static jint NativeCrypto_SSL_set_protocol_versions(JNIEnv* env, jclass, jlong ssl_address, CONSCRYPT_UNUSED jobject ssl_holder, jint min_version, jint max_version) {
+    CHECK_ERROR_QUEUE_ON_RETURN;
+    SSL* ssl = to_SSL(env, ssl_address, true);
+    JNI_TRACE("ssl=%p NativeCrypto_SSL_set_protocol_versions min=0x%x max=0x%x", ssl, min_version, max_version);
+    if (ssl == nullptr) {
+        return 0;
+    }
+    int min_result = SSL_set_min_proto_version(ssl, static_cast<uint16_t>(min_version));
+    int max_result = SSL_set_max_proto_version(ssl, static_cast<uint16_t>(max_version));
+    // Return failure if either call failed.
+    int result = std::min(min_result, max_result);
+    if (result == 0) {
```

Comment:
Hahaha. I would probably just have written:

```
if (!min_result || !max_result) {
   // failure case...
}
```

Reply:
Yeah, that's better.  I always default to thinking in terms of possibly getting -1 back with the OpenSSL APIs, even for the ones that just return 1 or 0.